### ISSUE TITLE
Glob order is non deterministic (cross-OS)

### ISSUE THREAD

**Issue by paolobrasolin** [ruby-analysis/delfos#37]
I'm using MacOS and the following specs are failing:

```
rspec ./lib/delfos/file_system/distance_calculation_spec.rb:41 # Delfos::FileSystem::DistanceCalculation#traversals should contain exactly "/tree/some_file", "/tree/even_more", and "/tree/another_top_level_file"
rspec ./lib/delfos/file_system/distance_calculation_spec.rb:56 # Delfos::FileSystem::DistanceCalculation#traversals should contain exactly "/tree/sub_directory/even_more" and "/tree/sub_directory/file_in_sub_directory"
rspec ./lib/delfos/file_system/relation_spec.rb:200 # Delfos::FileSystem::Relation traversing some files should contain exactly "/tree/sub_directory/even_more" and "/tree/sub_directory/file_in_sub_directory"
rspec ./lib/delfos/file_system/relation_spec.rb:212 # Delfos::FileSystem::Relation traversing up some files should contain exactly "/tree/sub_directory/some_more" and "/tree/sub_directory/file_in_sub_directory"
rspec ./lib/delfos/file_system/relation_spec.rb:232 # Delfos::FileSystem::Relation traversing up some directories should contain exactly "/tree/sub_directory/yet_another_directory" and "/tree/sub_directory/second_level"
rspec ./lib/delfos/file_system/relation_spec.rb:244 # Delfos::FileSystem::Relation traversing down all directories should contain exactly "/tree/sub_directory/another_directory", "/tree/sub_directory/second_level", and "/tree/sub_directory/yet_another_directory"
rspec ./lib/delfos/file_system/relation_spec.rb:257 # Delfos::FileSystem::Relation traversing down some directories should contain exactly "/tree/sub_directory/another_directory" and "/tree/sub_directory/second_level"
```

Here is the confusing result from my console when browsing `fixtures/tree/sub_directory`:

```
bash-3.2$ ruby -e "puts Dir.glob('*')"
another_directory
yet_another_directory
file_in_sub_directory
even_more
second_level
some_more

bash-3.2$ ls -l
total 0
drwxr-xr-x  3 paolobrasolin  staff  96 May 11 19:10 another_directory
-rw-r--r--  1 paolobrasolin  staff   0 May 11 19:10 even_more
-rw-r--r--  1 paolobrasolin  staff   0 May 11 19:10 file_in_sub_directory
drwxr-xr-x  3 paolobrasolin  staff  96 May 11 19:10 second_level
-rw-r--r--  1 paolobrasolin  staff   0 May 11 19:10 some_more
drwxr-xr-x  3 paolobrasolin  staff  96 May 11 19:10 yet_another_directory
```

This is all due to the fact that the order in which `Dir.glob` returns actually is OS-dependant, as stated in this reference: http://ruby-doc.org/core-2.5.0/Dir.html#method-c-glob

This is solved by #36

**Comment by paolobrasolin** [comment]
Here is a detailed log of the failures:

```
Failures:

  1) Delfos::FileSystem::DistanceCalculation#traversals should contain exactly "/tree/some_file", "/tree/even_more", and "/tree/another_top_level_file"
     Failure/Error: expect(a).to match_array(b.map { |f| t(f) }.map(&format))

       expected collection contained:  ["/tree/another_top_level_file", "/tree/even_more", "/tree/some_file"]
       actual collection contained:    ["/tree/another_top_level_file", "/tree/some_file"]
       the missing elements were:      ["/tree/even_more"]
     # ./spec/support/helper_methods.rb:30:in `match_file_array'
     # ./lib/delfos/file_system/distance_calculation_spec.rb:49:in `block (3 levels) in <module:FileSystem>'

  2) Delfos::FileSystem::DistanceCalculation#traversals should contain exactly "/tree/sub_directory/even_more" and "/tree/sub_directory/file_in_sub_directory"
     Failure/Error: expect(a).to match_array(b.map { |f| t(f) }.map(&format))

       expected collection contained:  ["/tree/sub_directory/even_more", "/tree/sub_directory/file_in_sub_directory"]
       actual collection contained:    ["/tree/sub_directory/even_more", "/tree/sub_directory/file_in_sub_directory", "/tree/sub_directory/some_more"]
       the extra elements were:        ["/tree/sub_directory/some_more"]
     # ./spec/support/helper_methods.rb:30:in `match_file_array'
     # ./lib/delfos/file_system/distance_calculation_spec.rb:59:in `block (3 levels) in <module:FileSystem>'

  3) Delfos::FileSystem::Relation traversing some files should contain exactly "/tree/sub_directory/even_more" and "/tree/sub_directory/file_in_sub_directory"
     Failure/Error: expect(a).to match_array(b.map { |f| t(f) }.map(&format))

       expected collection contained:  ["/tree/sub_directory/even_more", "/tree/sub_directory/file_in_sub_directory"]
       actual collection contained:    ["/tree/sub_directory/even_more", "/tree/sub_directory/file_in_sub_directory", "/tree/sub_directory/some_more"]
       the extra elements were:        ["/tree/sub_directory/some_more"]
     # ./spec/support/helper_methods.rb:30:in `match_file_array'
     # ./lib/delfos/file_system/relation_spec.rb:201:in `block (3 levels) in <top (required)>'

  4) Delfos::FileSystem::Relation traversing up some files should contain exactly "/tree/sub_directory/some_more" and "/tree/sub_directory/file_in_sub_directory"
     Failure/Error: expect(a).to match_array(b.map { |f| t(f) }.map(&format))

       expected collection contained:  ["/tree/sub_directory/file_in_sub_directory", "/tree/sub_directory/some_more"]
       actual collection contained:    ["/tree/sub_directory/even_more", "/tree/sub_directory/file_in_sub_directory", "/tree/sub_directory/some_more"]
       the extra elements were:        ["/tree/sub_directory/even_more"]
     # ./spec/support/helper_methods.rb:30:in `match_file_array'
     # ./lib/delfos/file_system/relation_spec.rb:213:in `block (3 levels) in <top (required)>'

  5) Delfos::FileSystem::Relation traversing up some directories should contain exactly "/tree/sub_directory/yet_another_directory" and "/tree/sub_directory/second_level"
     Failure/Error: expect(a).to match_array(b.map { |f| t(f) }.map(&format))

       expected collection contained:  ["/tree/sub_directory/second_level", "/tree/sub_directory/yet_another_directory"]
       actual collection contained:    ["/tree/sub_directory/second_level"]
       the missing elements were:      ["/tree/sub_directory/yet_another_directory"]
     # ./spec/support/helper_methods.rb:30:in `match_file_array'
     # ./lib/delfos/file_system/relation_spec.rb:233:in `block (3 levels) in <top (required)>'

  6) Delfos::FileSystem::Relation traversing down all directories should contain exactly "/tree/sub_directory/another_directory", "/tree/sub_directory/second_level", and "/tree/sub_directory/yet_another_directory"
     Failure/Error: expect(a).to match_array(b.map { |f| t(f) }.map(&format))

       expected collection contained:  ["/tree/sub_directory/another_directory", "/tree/sub_directory/second_level", "/tree/sub_directory/yet_another_directory"]
       actual collection contained:    ["/tree/sub_directory/another_directory", "/tree/sub_directory/yet_another_directory"]
       the missing elements were:      ["/tree/sub_directory/second_level"]
     # ./spec/support/helper_methods.rb:30:in `match_file_array'
     # ./lib/delfos/file_system/relation_spec.rb:245:in `block (3 levels) in <top (required)>'

  7) Delfos::FileSystem::Relation traversing down some directories should contain exactly "/tree/sub_directory/another_directory" and "/tree/sub_directory/second_level"
     Failure/Error: expect(a).to match_array(b.map { |f| t(f) }.map(&format))

       expected collection contained:  ["/tree/sub_directory/another_directory", "/tree/sub_directory/second_level"]
       actual collection contained:    ["/tree/sub_directory/another_directory", "/tree/sub_directory/second_level", "/tree/sub_directory/yet_another_directory"]
       the extra elements were:        ["/tree/sub_directory/yet_another_directory"]
     # ./spec/support/helper_methods.rb:30:in `match_file_array'
     # ./lib/delfos/file_system/relation_spec.rb:258:in `block (3 levels) in <top (required)>'
```

**Comment by markburns** [comment]
Fixed by #36
